Fixes #611 LSP4Jakarta ambiguously matches the expected class names when checking diagnostics, applying quick fixes, etc

aparnamichael opened the pull request:
Fixes #611

LSP4Jakarta ambiguously matches the expected class names when checking diagnostics, applying quick fixes, etc

Updated all of the diagnostic collectors to include fully qualified class names in the diagnostic data that they produce. And fixed the quick fix  issue of JsonbTransientAnnotationQuickFix and ScopeDeclarationQuickFix.

vaisakhkannan commented on src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbDiagnosticsCollector.java:
* Copyright (c) 2020, 2023 IBM Corporation, Matheus Cruz and others. 

Update the year to 2024

aparnamichael replied:
done

vaisakhkannan reviewed (dismissed):
looks good to me!

mrglavas commented on src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java:
```diff
@@ -108,12 +105,10 @@ else if (INJECT_FQ_NAME.equals(annotation))
-                    List<String> diagnosticData = fieldScopes.stream().map(annotation -> getSimpleName(annotation))
-                            .collect(Collectors.toList()); // convert to simple name
-                    diagnosticData.add(PRODUCES);
+                    fieldScopes.add(PRODUCES);
```
Ensure that all names in the list are qualified names.

mrglavas commented on src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java:
```diff
@@ -167,12 +162,10 @@ else if (INJECT_FQ_NAME.equals(annotation))
-                    List<String> diagnosticData = methodScopes.stream().map(annotation -> getSimpleName(annotation))
-                            .collect(Collectors.toList()); // convert to simple name
-                    diagnosticData.add(PRODUCES);
+                    methodScopes.add(PRODUCES);
```
Ensure that all names in the list are qualified names.

aparnamichael replied:
Done

mrglavas commented on src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java:
```diff
@@ -104,46 +104,92 @@ public void scopeDeclaration() throws Exception {
                 DiagnosticSeverity.Error, "jakarta-cdi", "InvalidScopeDecl");
-        d1.setData(new Gson().toJsonTree(Arrays.asList("Dependent", "ApplicationScoped", "Produces")));
+        d1.setData(new Gson().toJsonTree(Arrays.asList("jakarta.enterprise.context.Dependent",
+                "jakarta.enterprise.context.ApplicationScoped", "Produces")));
```
This expected output is incorrect. All class names in the list should be fully qualified.

aparnamichael replied:
Done

mrglavas commented on src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbTransientAnnotationQuickFix.java:
```diff
@@ -35,20 +36,15 @@ public class JsonbTransientAnnotationQuickFix extends RemoveMultipleAnnotations
-                fqAnnotations.addAll(getFQAnnotationNames(project, annotation));
-            }
-            annotationsListsToRemove.add(fqAnnotations);
+            annotationsListsToRemove.addAll(Collections.singleton(annotations));
```
This seems to do the same thing as the old line of code but is less efficient and harder to understand. Why wrap the list in a collection?

aparnamichael replied:
Done

aparnamichael commented on src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java:
```diff
@@ -104,46 +104,92 @@ public void scopeDeclaration() throws Exception {
+                        "by a producer method at most once.", DiagnosticSeverity.Error, "jakarta-cdi",
+                "InvalidScopeDecl");
+        d2.setData(new Gson().toJsonTree(Arrays.asList("jakarta.enterprise.context.ApplicationScoped",
+                "jakarta.enterprise.context.RequestScoped", "Produces")));
```
Done

vaisakhkannan reviewed (approved):
Looks good

aparnamichael commented:
@mrglavas I have addressed all the review comments.

Also I have gone through all of the diagnostic classes and all of them are child class of AbstractDiagnosticsCollector. And all of these classes are using fully qualified name.
